packages/logo-grid/partials/dialog-tooltip/index.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { useState, useRef, useEffect } from 'react'
+import { DialogOverlay, DialogContent } from '@reach/dialog'
+import VisuallyHidden from '@reach/visually-hidden'
+import { useRect } from '@reach/rect'
+import s from './style.module.css'
+import Portal from '@reach/portal'
+import InlineSvg from '@hashicorp/react-inline-svg'
+import svgX from '../../icons/x.svg?include'
+import classNames from 'classnames'
+
+interface DialogTooltipProps {
+  /** Elements to render in the content area of the dialog. */
+  children: React.ReactNode
+  /** Function to set the shown state of the dialog. Necessary for close functionality. */
+  setIsShown: React.Dispatch<React.SetStateAction<boolean>>
+  /** Whether to show the dialog or not. */
+  shown: boolean
+  /** DOMRect of the element that triggered the dialog. */
+  triggerRect: DOMRect
+  /** */
+  arrowSize?: number
+  /** Minimum distance in pixels that the dialog should be from the viewport edge.  */
+  collisionBuffer?: number
+  /** Color scheme appearance of the component. Works best in contexts with a matching theme. */
+  theme?: 'light' | 'dark'
+}
+
+function DialogTooltip({
+  children,
+  setIsShown,
+  shown,
+  triggerRect,
+  arrowSize = 10,
+  collisionBuffer = 8,
+  theme = 'light',
+}: DialogTooltipProps): React.ReactElement {
+  /* Forces update, need to fix, see useEffect comment below */
+  const [, setDate] = useState(0)
+  const setNow = () => setDate(Date.now())
+
+  const tooltipRef = useRef(null)
+  const tooltipRect = useRect(tooltipRef, { observe: true })
+
+  const position = calcDialogPosition(
+    triggerRect,
+    tooltipRect,
+    collisionBuffer,
+    arrowSize
+  )
+
+  /* Odd fix for possible issue with useRect, where
+  tooltipRect did not seem to be set as expected.
+  I think this works because it forces a re-render
+  when "shown" changes, which is exactly when we need
+  to look at the tooltipRef again and re-measure.
+  Without this, tooltipRect seems to always be null.
+  Not actually sure why this works. Need to investigate. */
+  useEffect(setNow, [shown])
```

Comment:
I need help understanding this 😅 It _feels_  like `useRect` should just work without this, and update properly with the dialog's `DOMRect` when it is opened (and therefore rendered). I'm not sure why  it  doesn't. Possibly related to `Portal` use, as  I did test the same thing without `Portal` and it worked fine. cc @BRKalow @jescalan @thiskevinwang in case y'all have any insight 🙇